src/library.cpp

```diff
@@ -200,6 +200,11 @@ std::string Library::getBestTargetBookId(const Bookmark& bookmark) const {
     return "";
   }
   if (targetBooks.size() != 1) {
+    // We have several, reduce to same flavour
+    auto flavouredTargetBooks = filter(book_filter.flavour(bookmark.getBookFlavour()));
+    if (!flavouredTargetBooks.empty()) {
+      targetBooks = flavouredTargetBooks;
+    }
```

Comment:
`flavouredTargetBooks` may contain `bookmark.getBookId()` (which has been removed from `targetBooks` and will now reappear there)